src/project-css/app/Helpers/CSVHelper.php

```diff
@@ -182,96 +188,61 @@ public function determineTypes($hasheader, $fltFleAbsPth, $readCount) {
          // determine final field types
          foreach($checkArray as $x=>$x_value)
          {
-           // if integer is detected and character count is greater than
-           // 10 we will store it as text
-           if (($x_value[0] == 1) && ($x_value[1] > 10)) {
-             $x_value[0] = 0;
-           }
-
-           if ($x_value[0] == 0) {
-             switch ($x_value[1]) {
-                 case 0:
-                 case $x_value[1] < 30:
-                     array_push($fieldType, ['string', 'default', $x_value[1]]);
-                     break;
-                 case $x_value[1] < 150:
-                     array_push($fieldType, ['string', 'medium', $x_value[1]]);
-                     break;
-                 case $x_value[1] < 500:
-                     array_push($fieldType, ['string', 'big', $x_value[1]]);
-                     break;
-                 case $x_value[1] < 2000:
-                     array_push($fieldType, ['text', 'default', $x_value[1]]);
-                     break;
-                 case $x_value[1] < 8000:
-                     array_push($fieldType, ['text', 'medium', $x_value[1]]);
-                     break;
-                 default:
-                     array_push($fieldType, ['text', 'big', $x_value[1]]);
-             }
-           }
-           elseif ($x_value[0] == 1) {
-             switch ($x_value[1]) {
-                 case $x_value[1] <= 2:
-                   array_push($fieldType, ['integer', 'default', $x_value[1]]);
-                   break;
-                 case $x_value[1] <= 7:
-                   array_push($fieldType, ['integer', 'medium', $x_value[1]]);
-                   break;
-                 default:
-                     array_push($fieldType, ['integer', 'big', $x_value[1]]);
-             }
-           }
+          if ($x_value[0] == 0) {
```

Comment:
The logical complexity here just blows up all mathematical calculations. 

You are determining how big each string is and what the character values need to be, as well as determine what integer values need to be, I think by FAR this is doing too much in a single function. You should break this down into multiple methods or another class call. 

I think if it was me I would create library or class module that has methods to handle all of this switch return value information.